### ISSUE TITLE
TypeError when filtering transactions with Advanced Filters by Date 

### ISSUE THREAD

**Issue by jacoswan** [Automattic/woocommerce-payments#3610]
### Describe the bug
When attempting to filter transactions with Advanced Filters > Date, clicking on the calendar to change months causes the following error that prevents any results from showing:

> There was an error rendering this view. Please contact support for assistance if the problem persists.
TypeError: t.onMonthPreviewed is not a function

It happens regardless of the setting of the Before | After | Between dropdown. 

Workaround: manually enter the dates, which still works. 

### To Reproduce

1. Go to WP Admin > Payments > Transactions
2. Click All Transactions > Advanced Filters > Add Filter > Date
3. Click in the calendar field and try to change the month
4. Error occurs

### Actual behavior
The error occurs and the view is no longer operable. 

### Screenshots

![Screen Capture on 2022-01-07 at 14-25-39](https://user-images.githubusercontent.com/15178758/148557831-7a3ac005-e1af-409e-92c2-342b25025d6b.gif)

### Expected behavior
Should be able to use the calendar selector to select date ranges. 

### Desktop (please complete the following information):

* OS: Mac OS 11.6.1 (20G224)
* Browser: Chrome Version 96.0.4664.93

### Additional context

As per 4667708-zen

**Comment by brianyu28** [comment]
This looks like it's caused by a bug in Gutenberg that was fixed in https://github.com/WordPress/gutenberg/pull/31751 for Gutenberg 10.9. The fix isn't included in WordPress core yet, but I expect it should be in WordPress 5.9, [to be released 25 January 2022](https://make.wordpress.org/core/5-9/).

As a temporary workaround, manual date entry does work (as you note), and merchants can also install the latest version of Gutenberg directly, which I just tested and confirmed does fix the problem.

I'll close the issue for now, but we should reopen if the problem persists after the WordPress 5.9 release.

--

After investigating this, I also realized that the issue has been raised before in #2507, where the same conclusion was reached.
